web/src/components/people/AdminView.tsx

```diff
@@ -1,80 +1,156 @@
+import Loading from "../ui/Loading";
 
 const CheckIcon = dynamic(() => import("uikit-react-public").then((mod) => mod.Icon.Check), {
   ssr: false,
 });
-const XIcon = dynamic(() => import("uikit-react-public").then((mod) => mod.Icon.X), {
-  ssr: false,
-});
 
 function convertRFC3339ToDDMMYYYY(dateString: string) {
   const date = new Date(dateString);
   const day = String(date.getDate()).padStart(2, "0");
-  const month = String(date.getMonth() + 1).padStart(2, "0"); // January is 0!
+  const month = String(date.getMonth() + 1).padStart(2, "0");
   const year = date.getFullYear();
 
   return `${day}/${month}/${year}`;
 }
 
+function getHumanReadableTrainingKind(trainingKind: string) {
+  // getting the key from the value
+  const humanReadableTrainingKind = Object.entries(TrainingKindOptions).find(
+    ([, value]) => value === trainingKind
+  )?.[0];
+
+  return humanReadableTrainingKind;
+}
+
 export default function AdminView() {
   const [people, setPeople] = useState<People | null>(null);
+  const [trainingDialogOpen, setTrainingDialogOpen] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [id, setId] = useState("");
 
   useEffect(() => {
     const fetchPeople = async () => {
+      setIsLoading(true);
       try {
         const response = await getPeople();
-
         if (response.response.ok && response.data) {
           setPeople(response.data as People);
         }
       } catch (error) {
         console.error("Failed to get people:", error);
         setPeople(null);
+      } finally {
+        setIsLoading(false);
       }
     };
     fetchPeople();
   }, []);
 
+  const updatePersonUI = (id: string, training: TrainingRecord) => {
+    // get people object and find the person with the right id then update the training
+    if (!people) return;
+
+    const updatedPeople = [...people];
+    const personIndex = updatedPeople.findIndex((person) => person.user.id === id);
+
+    if (personIndex !== -1) {
+      const person = updatedPeople[personIndex];
+      person.training_record.training_records ??= [];
+      const recordIndex = person.training_record.training_records.findIndex((record) => record.kind === training.kind);
+
+      // If the training record exists, remove it
+      if (recordIndex !== -1) {
+        person.training_record.training_records.splice(recordIndex, 1);
+      }
+
+      person.training_record.training_records.push(training);
+      setPeople(updatedPeople);
+    }
+  };
+
   if (!people) return null;
 
+  const handleEditTrainingClick = (id: string) => {
+    setId(id);
+    setTrainingDialogOpen(true);
+  };
+
+  if (isLoading) {
+    return (
+      <div className={styles.container}>
+        <h2>Your Tasks</h2>
+        <Loading message="Loading users..." />
+      </div>
+    );
+  }
   return (
     <>
+      {trainingDialogOpen && (
+        <TrainingForm id={id} setTrainingDialogOpen={setTrainingDialogOpen} updatePersonUI={updatePersonUI} />
+      )}
```

Comment:
You could simplify this so that the component gets rendered based on the `id` being set:

```
{selectedUserId && (
        <TrainingForm 
          id={selectedUserId} 
          setTrainingDialogOpen={() => setSelectedUserId(null)} 
          updatePersonUI={updatePersonUI} 
        />
      )}
```

That way, you wouldn't need the additional `setTrainingDialogOpen`.

Up to you if you want to implement it this way or not, I don't feel strongly about it.